Userland/Libraries/LibJS/Runtime/DatePrototype.cpp

```diff
@@ -144,6 +148,48 @@ JS_DEFINE_NATIVE_FUNCTION(DatePrototype::get_hours)
     return Value(static_cast<double>(this_object->hours()));
 }
 
+JS_DEFINE_NATIVE_FUNCTION(DatePrototype::set_hours)
+{
+    auto* this_object = typed_this(vm, global_object);
+    if (!this_object)
+        return {};
+
+    auto new_hours = vm.argument(0).to_i32(global_object);
```

Comment:
this should use `to_number()` - `to_i32()` will hide NaN and Infinity, but we don't have invalid dates yet so fine for now I guess